indico/modules/events/registration/controllers/display.py

```diff
@@ -566,3 +580,65 @@ def _process_args(self):
+        RHParticipantList._check_access(self)
+        if registration_settings.get(self.event, 'merge_registration_forms'):
+            if 'picture' not in registration_settings.get(self.event, 'participant_list_columns'):
+                raise Forbidden
+            if not self.data.field_data.field.personal_data_type:
+                # only main picture from personal data is in merged form
+                raise Forbidden
+        else:
+            participant_list_form_columns = registration_settings.get(self.event, 'participant_list_form_columns')
+            if (self.data.field_data.field_id not in
+                    participant_list_form_columns[str(request.view_args['reg_form_id'])]):
+                raise Forbidden
+        is_participant = self.registration.event.is_user_registered(session.user)
+        if not self.registration.is_publishable(is_participant):
+            raise Forbidden
+
+    def _process_args(self):
+        RHParticipantList._process_args(self)
+        try:
+            self.data = (RegistrationData.query
+                         .filter(RegistrationData.field_data_id == request.view_args['field_data_id'],
+                                 RegistrationData.registration_id == request.view_args['registration_id'],
+                                 RegistrationFormItem.input_type == 'picture',
+                                 Registration.event_id == request.view_args['event_id'],
+                                 Registration.registration_form_id == request.view_args['reg_form_id'],
+                                 ~Registration.is_deleted,
+                                 ~RegistrationForm.is_deleted)
+                         .join(RegistrationData.field_data)
+                         .join(RegistrationFormFieldData.field)
+                         .join(RegistrationData.registration)
+                         .join(Registration.registration_form)
+                         .options((joinedload('registration').load_only(Registration.id,
+                                                                        Registration.consent_to_publish,
+                                                                        Registration.participant_hidden,
+                                                                        Registration.state,
+                                                                        Registration.is_deleted))
+                                  .joinedload('registration_form')
+                                  .load_only(RegistrationForm.id,
+                                             RegistrationForm.publish_registrations_participants,
+                                             RegistrationForm.publish_registrations_public,
+                                             RegistrationForm.publish_registrations_duration),
+                                  lazyload('*'))
+                         .one())
+        except NoResultFound:
+            raise BadRequest
```

Comment:
Is that actually needed? NoResultFound is already converted to a NotFound (404) error in the base RH code.

Reply:
Removed.